src/ext/sse.js

```diff
@@ -148,6 +148,7 @@ This extension adds support for Server Sent Events to htmx.  See /www/extensions
 					}
 
 					// swap the response into the DOM and trigger a notification
+					api.triggerEvent(elt, "htmx:sseBeforeMessage", event);
```

Comment:
This probably should provide cancellation support by wrapping this call in an if statement, like so 
https://github.com/bigskysoftware/htmx/blob/master/src/ext/ws.js#L131-L136

Also, would be nice to add tests for this event (and its cancellation)